closed/src/java.base/share/native/libjli/criuhelpers.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2024, 2024 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+#ifndef CRIUHELPERS_H
+#define CRIUHELPERS_H
+
+#include <assert.h>
+
+#include "java.h"
+#include "jni.h"
+#include "j9cfg.h"
+#if defined(J9VM_OPT_CRAC_SUPPORT)
+#include <ctype.h>
+#include <sys/wait.h>
+
+static const char *
+getCommandLineOptionValue(const char *optionName, int argc, char **argv, int *error);
+
+static const char *
+getCheckpointDirectory(int argc, char **argv, int *error);
+
+static int
+getLogLevel(int argc, char **argv, int *error);
+
+static jboolean
+isUnprivilegedModeOn(int argc, char **argv, int *error);
+
+static const char *
+getLogFile(int argc, char **argv, int *error);
+
+static int
+restoreFromCheckpoint(const char *checkpointDirectory, int logLevel, jboolean unprivilegedModeOn, const char *logFile);
+
+void
+handleCRaCRestore(int argc, char **argv);
+#endif /* defined(J9VM_OPT_CRAC_SUPPORT) */
```

Comment:
Header files should not declare `static` features: they belong in the relevant C/C++ source file.
This should include only what is necessary for the declarations here.
The result should be just:
```c
#include "j9cfg.h"

#if defined(J9VM_OPT_CRAC_SUPPORT)
void
handleCRaCRestore(int argc, char **argv);
#endif /* defined(J9VM_OPT_CRAC_SUPPORT) */
```

Reply:
How can I keep the includes for `java.h`, `<ctype.h>` and `<sys/wait.h>` and avoid the compilation errors that result when I keep just 

```c
#include "j9cfg.h"

#if defined(J9VM_OPT_CRAC_SUPPORT)
void
handleCRaCRestore(int argc, char **argv);
#endif /* defined(J9VM_OPT_CRAC_SUPPORT) */
```

?